typescript.js

```diff
@@ -13,7 +13,11 @@ module.exports = {
         // Allow the use of underscore to show args are not used.
         // This is helpful for seeing that a function implements
         // an interface but won't be using one of it's arguments.
-        "@typescript-eslint/no-unused-vars": ["error", { "args": "none" }],
+        "@typescript-eslint/no-unused-vars": ["error", {
+            "args": "none",
+            "destructuredArrayIgnorePattern": "^_",
+            "ignoreRestSiblings": true,
```

Comment:
I like a more explicit pattern for this:
```
const foo = { a: 1, b: 2, c: 3 };
// explicitly ignore a
const { a: _a, ...rest };

// do something with rest
```

```
"@typescript-eslint/no-unused-vars": ["error", {
                    "args": "none",
                    "ignoreRestSiblings": false,
                    "varsIgnorePattern": "^_",
                }],
```

Reply:
I agree it'd be good to require all unused destructuring variables to begin with an `_`, but in the internal discussion, it sounded like at least @turt2live was opposed to allowing unused `_vars` in the general case when we're not destructuring things. It's a shame there's no `restSiblingsIgnorePattern` option, but this may be the narrowest change it's possible for us to make.

Reply:
I have no strong opinions on this. Ignoring underscore prefixed variables in general seems sane, but if we use any underscore-prefixed variables for real this sounds like a thing we should fix anyway (and shouldn't be that hard..?)

Also, to note from web weekly, we think you should also be able to just omit unused array members in an array destructure (ie. `const [,,baz] = foobarbaz;`) (although this doesn't mean our codebase currently does of course).

Reply:
Doing a quick grep, it does look like omitting unused array members is the more common pattern in our codebase, so I can remove `destructuredArrayIgnorePattern` from this PR and just edit existing instances to use that pattern. There are a lot of places where we're currently disabling `no-unused-vars` or setting it to `"ignoreRestSiblings": true` to deal with object destructures, though, so I think `ignoreRestSiblings` is still worth adding.

Reply:
Updated, now I'm only adding `ignoreRestSiblings`.